tools/webdriver/webdriver/client.py

```diff
@@ -218,7 +218,9 @@ def perform(self, actions=None):
                         ``ActionSequence.dict``.
         """
         body = {"actions": [] if actions is None else actions}
-        return self.session.send_session_command("POST", "actions", body)
+        actions = self.session.send_session_command("POST", "actions", body)
+        self.session.switch_frame("parent")
```

Comment:
I'm wondering why we switch to the parent here. Is it so that future commands are sent to the right context? That seems like it might be brittle if you have more than one level of nested frames.

I think a comment would be good here either way.

Reply:
The reason I switch to parent is I was keeping getting this error
ERROR [expected OK] /pointerevents/pointerevent_attributes_nohover_pointers.html
└ → javascript error: window.__wptrunner_process_next_event is not a function

jgraham@ commented:
The window.__wptrunner_process_next_event error comes from the fact that in order to spin the event loop driving the tests we need to execute script in the top level window. So you need to ensure that the WebDriver window is set to the top level window whenever we try to do that.

Actually for one action call, we cannot support more than one frame. Maybe we can send them each time.

Reply:
Thanks for the context. Lets expand the comment a bit to mention that we can only go one frame deep, so "parent" is sufficient. Thanks!